packages/remix-dev/compiler/compileBrowser.ts

```diff
@@ -123,29 +152,108 @@ export const createBrowserCompiler = (
   remixConfig: RemixConfig,
   options: CompileOptions
 ): BrowserCompiler => {
-  let compiler: esbuild.BuildIncremental;
-  let esbuildConfig = createEsbuildConfig(remixConfig, options);
+  let appCompiler: esbuild.BuildIncremental;
+  let cssCompiler: esbuild.BuildIncremental;
+
   let compile = async (manifestChannel: WriteChannel<AssetsManifest>) => {
-    let metafile: esbuild.Metafile;
-    if (compiler === undefined) {
-      compiler = await esbuild.build({
-        ...esbuildConfig,
-        metafile: true,
-        incremental: true,
-      });
-      invariant(compiler.metafile, "Expected metafile to be defined");
-      metafile = compiler.metafile;
-    } else {
-      let rebuild = await compiler.rebuild();
-      invariant(rebuild.metafile, "Expected metafile to be defined");
-      metafile = rebuild.metafile;
-    }
-    let manifest = await createAssetsManifest(remixConfig, metafile);
+    let appBuildTask = async () => {
+      appCompiler = await (!appCompiler
+        ? esbuild.build({
+            ...createEsbuildConfig("app", remixConfig, options),
+            metafile: true,
+            incremental: true,
+          })
+        : appCompiler.rebuild());
+
+      invariant(
+        appCompiler.metafile,
+        "Expected app compiler metafile to be defined"
```

Comment:
```suggestion
        "Expected app compiler metafile to be defined. This is likely a bug in Remix. Please open an issue at https://github.com/remix-run/remix/issues/new"
```

Reply:
This was an existing error message—this PR just reformatted it—but I agree with your improvement 👍